scripts/sanity_chk/sanitylib.py

```diff
@@ -3502,7 +3502,7 @@ def json_report(self, filename, platform=None, append=False, version="NA"):
                 for i in data["test_suite"]:
                     test_details = i["test_details"]
                     for test_data in test_details:
-                        if not (test_data["status"]) == "failed":
+                        if not (test_data.get("status")) == "failed":
```

Comment:
this conditional is awkward, why not 

```
if test.data.get(...) != "failed"
```
or whatever you are trying to check for here. It will still fail otherwise I think.